server/src/main/java/org/opensearch/search/backpressure/stats/SearchBackpressureTaskStats.java

```diff
@@ -85,10 +85,25 @@ public void writeTo(StreamOutput out) throws IOException {
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-        SearchShardTaskStats that = (SearchShardTaskStats) o;
+        SearchBackpressureTaskStats that = (SearchBackpressureTaskStats) o;
         return cancellationCount == that.cancellationCount
             && limitReachedCount == that.limitReachedCount
-            && resourceUsageTrackerStats.equals(that.resourceUsageTrackerStats);
+            && compareMaps(resourceUsageTrackerStats, that.resourceUsageTrackerStats);
```

Comment:
Curious to know why this change was required?

Reply:
ok I added this for testing purpose. Some test was failing, but missed removing it while raising the PR. I will revert it to `resourceUsageTrackerStats.equals(that.resourceUsageTrackerStats);` in the next commit.